src/DynamoRevit/DynamoRevitApp.cs

```diff
@@ -117,6 +165,34 @@ private static void RegisterAdditionalUpdaters(UIControlledApplication applicati
             Updaters.Add(sunUpdater);
         }
 
+        private void SubscribeApplicationEvents()
+        {
+            UIControlledApplication.Idling += OnApplicationIdle;
+
+            proxy = new EventHandlerProxy();
+
+            UIControlledApplication.ViewActivated += proxy.OnApplicationViewActivated;
+            UIControlledApplication.ViewActivating += proxy.OnApplicationViewActivating;
+
+            ControlledApplication.DocumentClosing += proxy.OnApplicationDocumentClosing;
+            ControlledApplication.DocumentClosed += proxy.OnApplicationDocumentClosed;
+            ControlledApplication.DocumentOpened += proxy.OnApplicationDocumentOpened;
+        }
+
+        private void UnsubscribeApplicationEvents()
+        {
+            UIControlledApplication.Idling -= OnApplicationIdle;
+
+            UIControlledApplication.ViewActivated -= proxy.OnApplicationViewActivated;
+            UIControlledApplication.ViewActivating -= proxy.OnApplicationViewActivating;
+
+            ControlledApplication.DocumentClosing -= proxy.OnApplicationDocumentClosing;
+            ControlledApplication.DocumentClosed -= proxy.OnApplicationDocumentClosed;
+            ControlledApplication.DocumentOpened -= proxy.OnApplicationDocumentOpened;
+
+            proxy = null;
```

Comment:
The proxy's lifetime is as long as the addin application, but not the Revit application.

Related to adding event handlers in the idling event handlers, it is not necessary to add event handlers for our own events in the idling event handler. I will make the change.